File: src/test/java/com/github/yonaprojects/yona/PasswordGenerator.java
package com.github.yonaprojects.yona;

import java.security.MessageDigest;
import java.util.Base64;
import java.nio.charset.StandardCharsets;

public class PasswordGenerator {
    public static void main(String[] args) throws Exception {
        String password = "1234";
        String salt = "f322603b";
        
        MessageDigest digest = MessageDigest.getInstance("SHA-256");
        digest.reset();
        digest.update(salt.getBytes(StandardCharsets.UTF_8));
        byte[] hashed = digest.digest(password.getBytes(StandardCharsets.UTF_8));
        for (int i = 1; i < 1024; i++) {
            digest.reset();
            hashed = digest.digest(hashed);
        }
        String result = Base64.getEncoder().encodeToString(hashed);
        System.out.println("GENERATED_PASSWORD_HASH:" + result);
    }
}

File: src/test/kotlin/com/github/yonaprojects/yona/PasswordGenerator.kt
package com.github.yonaprojects.yona

import java.security.MessageDigest
import java.util.Base64

fun main() {
    val password = "1234"
    val salt = "f322603b"
    
    val digest = MessageDigest.getInstance("SHA-256")
    digest.reset()
    digest.update(salt.toByteArray(Charsets.UTF_8))
    var hashed = digest.digest(password.toByteArray(Charsets.UTF_8))
    for (i in 1 until 1024) {
        digest.reset()
        hashed = digest.digest(hashed)
    }
    val result = Base64.getEncoder().encodeToString(hashed)
    println("GENERATED_PASSWORD_HASH:$result")
}
